podcastfy/core/podcast.py

```diff
@@ -141,25 +94,22 @@ def wrapper(self, *args, **kwargs):
 
 
 class Podcast:
-    def __init__(self, urls: List[str], llm_backend: LLMBackend, tts_backend: TTSBackend,
-                 parsing_backend: ParserCollection):
-        self.urls = urls
+    def __init__(self, content: str, llm_backend: LLMBackend, tts_backend: TTSBackend):
```

Comment:
There should be a Class Transcript and PodcastAudio, and Podcast should contain both. So its init would have

self.transcript = Transcript()
self.audio = PodcastAudio()

in self.transcript the user would be able to access attributes such as transcript language, style, length etc
in self.audio the user would be able to access attributes such as audio language, length, format etc

What do you think?

Reply:
Just to be sure I understand, these attributes would then be used to generate the text or audio content ?
If yes, that was not really the idea behind this design which is quite low level and generic. To get back to the style things, they would be used to build the tts_backend I think. But other classes would help to make this instanciation easier. Maybe if you agree with the overall design, I should try soon to bridge your code with mine to see if it works ?

But here, the user car certainly edit the transcript or the audio segments then proceed to the next build stage.

I think I should have self.transcript = None and self.audio_segments=None and self.assembled_audio=None at the init time though.

assembled is maybe not a great name, it's the final audio in contrast with audio_segments, so maybe just audio is better.